### PR TITLE
Apigateway stage import

### PR DESCRIPTION
**Pull request by ericfode**
importer for apigateway_stage 
this allows us to work around #420

### CODE REVIEW

**Comment by Ninir** [comment]
Hi @ericfode 

This is getting into shape! is it still a WIP? do you need any help?

Thanks! :)

**Comment by ericfode** [comment]
No, I don't think it is. I would appreciate a code review. 
Currently, my only concern with this PR is around dealing with tags on API Gateway stages. Both the stage and the deployment seem to modify the tags. The changes I added seem to make it more deterministic, but it's nothing I'm proud of.  Any suggestions are appreciated.  @Ninir

File: aws/import_aws_api_gateway_stage_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAWSApiGatewayStage_importBasic(t *testing.T) {
+	resourceName := "tf-acc-test.dev"
```
Comment (bflad) [inline aws/import_aws_api_gateway_stage_test.go]: This should be a reference to a resource in a test configuration, for `testAccAWSAPIGatewayStageConfig_basic` its resource definition is:

```hcl
resource "aws_api_gateway_stage" "test" {
  rest_api_id = "${aws_api_gateway_rest_api.test.id}"
  stage_name = "prod"
  deployment_id = "${aws_api_gateway_deployment.dev.id}"
  cache_cluster_enabled = true
  cache_cluster_size = "0.5"
  variables {
    one = "1"
    two = "2"
  }
  tags {
    Name = "tf-test"
  }
}
```

So this should be `resourceName := "aws_api_gateway_stage.test"`.

It also looks that we require name randomization for `testAccAWSAPIGatewayStageConfig_basic()` now. You can create a variable `rName := acctest.RandomWithPrefix("tf-acc-test")` and call that function as `testAccAWSAPIGatewayStageConfig_basic(rName)`

File: aws/resource_aws_api_gateway_stage.go [hunk 2]
```diff
@@ -300,6 +304,12 @@ func diffVariablesOps(prefix string, oldVars, newVars map[string]interface{}) []
 			if oldValue == newValue {
 				continue
 			}
+		} else {
```
Comment (bflad) [inline aws/resource_aws_api_gateway_stage.go]: Can you explain this change here?

File: aws/resource_aws_api_gateway_stage.go [hunk 3]
```diff
@@ -18,6 +19,9 @@ func resourceAwsApiGatewayStage() *schema.Resource {
 		Read:   resourceAwsApiGatewayStageRead,
 		Update: resourceAwsApiGatewayStageUpdate,
 		Delete: resourceAwsApiGatewayStageDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsApiGatewayStageImportState,
```
Comment (bflad) [inline aws/resource_aws_api_gateway_stage.go]: To support import when the read function is not using `d.Id()` directly, we just need to simply `d.Set()` the attributes required for read. I would also recommend splitting on `/` which matches the format of the execution ARN. Something like this should work:

```go
State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
	idParts := strings.Split(d.Id(), "/")
	if len(idParts) != 2 || len(d.Id()) < 3 {
		return []*schema.ResourceData{}, fmt.Errorf("Wrong format of ID: %s. Please follow 'rest-api-id/stage-name'", d.Id())
	}
	d.Set("rest_api_id", idParts[0])
	d.Set("stage_name", idParts[1])
	return []*schema.ResourceData{d}, nil
},
```

**Comment by bflad** [comment]
Closing due to inactivity -- happy to review again if someone addresses the feedback.

**Pull request closed without merging**